File: Freedom/主界面/SettingsViewController.m
//  SettingsViewController.m
//  Freedom
//  Created by Super on 16/6/24.
//  Copyright © 2016年 Super. All rights reserved.
#import "SettingsViewController.h"
@interface SettingsViewController ()<UITableViewDelegate,UITableViewDataSource>{
    UITableView *tableView1;
}
@end
@implementation SettingsViewController
@synthesize contentLength, dismissByBackgroundDrag, dismissByBackgroundTouch, dismissByForegroundDrag;
- (void)viewDidLoad {
    [super viewDidLoad];
    tableView1 = [[UITableView alloc]initWithFrame:CGRectMake(0, 0, APPW, APPH)];
    tableView1.delegate = self;
    tableView1.dataSource = self;
    tableView1.backgroundColor = [UIColor clearColor];
    tableView1.tableHeaderView = [self getheadView];
    tableView1.showsVerticalScrollIndicator = NO;
    tableView1.separatorStyle = UITableViewCellSeparatorStyleNone;
    [self.view addSubview:tableView1];
    ElasticTransition *ET = (ElasticTransition*)self.transitioningDelegate;
    DLog(@"\ntransition.edge = %@\ntransition.transformType = %@\ntransition.sticky = %@\ntransition.showShadow = %@", [HelperFunctions typeToStringOfEdge:ET.edge], [ET transformTypeToString], ET.sticky ? @"YES" : @"NO", ET.showShadow ? @"YES" : @"NO");
}
-(id)initWithCoder:(NSCoder *)aDecoder{
    self = [super initWithCoder:aDecoder];
    if (self) {
        self.contentLength = APPW*0.8;
        self.dismissByBackgroundTouch   = YES;
        self.dismissByBackgroundDrag    = YES;
        self.dismissByForegroundDrag    = YES;
    }return self;
}
-(UIView *)getheadView{
    UIView *headView = [[UIView alloc]initWithFrame:CGRectMake(0, 0, APPW*0.8, 200)];
    return headView;
}
#pragma mark - Table view data source
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 20;
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];
    if(!cell){
        cell = [[UITableViewCell alloc]initWithFrame:CGRectMake(0, 0, 300, 100)];
    }
    cell.backgroundColor = [UIColor clearColor];
    cell.textLabel.text = [NSString stringWithFormat:@"%ld",indexPath.row];
    return cell;
}
-(CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    return 80;
}
-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    [tableView deselectRowAtIndexPath:indexPath animated:NO];
}
@end

File: Freedom/主界面/SettingsViewController.swift
//
//  SettingsViewController.swift
//  Freedom
//
//  Created by Super on 6/15/18.
//  Copyright © 2018 Super. All rights reserved.
//
import UIKit
import XExtension
import XCarryOn
class SettingsViewController: BaseViewController ,ElasticMenuTransitionDelegate{
    var contentLength:CGFloat = APPW * 0.8
    var dismissByBackgroundTouch = true
    var dismissByBackgroundDrag = true
    var dismissByForegroundDrag = true
    override func viewDidLoad() {
        super.viewDidLoad()
        tableView = BaseTableView(frame: CGRect(x: 0, y: 0, width: APPW, height: APPH))
//        tableView.delegate = self
//        tableView.dataSource = self
        tableView.backgroundColor = UIColor.clear
        tableView.tableHeaderView = UIView(frame: CGRect(x: 0, y: 0, width: APPW * 0.8, height: 200))
        tableView.showsVerticalScrollIndicator = false
        tableView.separatorStyle = .none
        view.addSubview(tableView)
        let tm = transitioningDelegate as! ElasticTransition
        print( "transition.edge = .\(tm.edge)\n" +
            "transition.transformType = .\(tm.transformType)\n" +
            "transition.sticky = \(tm.sticky)\n" +
            "transition.showShadow = \(tm.showShadow)")
    }
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 20
    }
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        var cell = tableView.dequeueCell(BaseTableViewCell.self)
        cell.backgroundColor = UIColor.clear
        cell.textLabel?.text = "\(indexPath.row)"
        return cell
    }
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 80
    }
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: false)
    }
}
